metrics/models.go

```diff
@@ -237,7 +256,16 @@ func (metadata *TestRunsMetadata) LoadTestRuns(ctx context.Context) (err error)
 // LoadTestRuns fetches the TestRun entities for the PassRateMetadata's TestRunIDs.
 func (metadata *TestRunsMetadataLegacy) LoadTestRuns(ctx context.Context) (err error) {
 	if len(metadata.TestRuns) == 0 {
-		metadata.TestRuns, err = metadata.TestRunIDs.LoadTestRuns(ctx)
+		newRuns, err := metadata.TestRunIDs.LoadTestRuns(ctx)
+		if err != nil {
+			return err
+		}
+		// Serialize + unserialize - aka "Copy fields"
+		serialized, err := json.Marshal(newRuns)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(serialized, &metadata.TestRuns)
```

Comment:
Just curious: is there a better way to copy structs and ignore non-existent fields than JSON roundtrip?

Reply:
Not that I'm aware. Reflection is verbose in Golang.